modules/experimental-layers/src/gpu-screen-grid-layer/gpu-screen-grid-layer-vertex.glsl.js

```diff
@@ -37,16 +38,49 @@ uniform AggregationData
   vec4 maxCount;
 } aggregationData;
 
+uniform vec4 colorRange[RANGE_COUNT];
+uniform vec2 colorDomain;
+uniform bool shouldUseMinMax;
+
 out vec4 vColor;
 
+vec4 quantizeScale(vec2 domain, vec4 range[RANGE_COUNT], float value) {
+  vec4 outColor = vec4(0., 0., 0., 0.);
+  if (value >= domain.x && value <= domain.y) {
+    float domainRange = domain.y - domain.x;
+    if (domainRange <= 0.) {
+      outColor = colorRange[0];
+    } else {
+      float rangeCount = float(RANGE_COUNT);
+      float rangeStep = domainRange / rangeCount;
+      float idx = floor((value - domain.x) / rangeStep);
+      idx = min(idx, rangeCount - 1.);
```

Comment:
will `clamp` give you the exact result in one line? 
`clamp(idx, 0., rangCount - 1.)`

Reply:
Yep, replace min, max with clamp, thanks !!